### PR TITLE
Redux impementation

### PR DESCRIPTION
**Pull request by ZhyrovDmytro**

### CODE REVIEW

File: src/app/components/Search.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 // Utilities
 import axios from 'axios';
 import React, { Component } from 'react';
+import requestServiceRedux from '../action-creators/requestService';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
```
Comment (fidelman) [inline src/app/components/Search.js]: unused var

File: src/app/components/Search.js [hunk 2]
```diff
@@ -1,6 +1,9 @@
 // Utilities
 import axios from 'axios';
 import React, { Component } from 'react';
+import requestServiceRedux from '../action-creators/requestService';
+import { connect } from 'react-redux';
```
Comment (fidelman) [inline src/app/components/Search.js]: absolute path first

File: src/app/store/configureStore.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import rootReducer from './root-reducer';
+
+export default function configureStore(initialState) {
+    /* eslint-disable no-underscore-dangle */
+    return createStore(
+        rootReducer,
+        initialState,
```
Comment (fidelman) [inline src/app/store/configureStore.js:10]: connect logger please

File: src/app/components/Search.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 // Utilities
 import axios from 'axios';
 import React, { Component } from 'react';
+import requestServiceRedux from '../action-creators/requestService';
```
Comment (fidelman) [inline src/app/components/Search.js]: Add prop types to every component

**Review by fidelman — CHANGES_REQUESTED** [review]

File: src/app/components/Search.js [hunk 5]
```diff
@@ -100,12 +100,12 @@ class Search extends Component {
...
 
     render() {
-        const { isFetching, images, existMoreItems } = this.props.requestService;
+        const { isFetching, images, existMoreItems, noImages } = this.props.requestService;
         const loader = isFetching && <Loader />;
-        // const noImages = this.props.noImages === true &&
-        //     (<div className="text-center mt-3">
-        //         <p>No images to display :(</p>
-        //     </div>);
+        const noImagesToShow = noImages &&
```
Comment (fidelman) [inline src/app/components/Search.js]: is it necessary to create a state for it?
can you check noImages status by `!images.length`?

**Review by fidelman — CHANGES_REQUESTED** [review]

File: src/app/action-creators/loadMore.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+        axios.get(path)
+            .then(response => {
+                if (response.data instanceof Array) {
+                    dispatch(getRandomImages(response.data));
+                } else if (response.data instanceof Object) {
+                    dispatch(getImages(response.data.results));
+                }
+            })
+            .catch((error) => {
+                console.log('error');
```
Comment (fidelman) [inline src/app/action-creators/loadMore.js]: it will be complicated to debug just to see the message `error`, maybe passe the error variable?

**Review by fidelman — CHANGES_REQUESTED** [review]

File: src/app/store/configureStore.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import thunk from 'redux-thunk';
+import logger from 'redux-logger';
+import rootReducer from './root-reducer';
+
+export default function configureStore(initialState) {
+    /* eslint-disable no-underscore-dangle */
+    return createStore(
+        rootReducer,
+        initialState,
+        applyMiddleware(thunk)
```
Comment (fidelman) [inline src/app/store/configureStore.js]: logger??? where are you??

**Review by fidelman — CHANGES_REQUESTED** [review]

File: src/app/components/Search.js (hunk 5 above)
Comment (ZhyrovDmytro) [reply src/app/components/Search.js]: My initial state has images.length === 0, so this message will be appear when page is loaded

**Pull request merged**